src/Riok.Mapperly/Symbols/MapperMethodReference.cs

```diff
@@ -0,0 +1,5 @@
+﻿using Microsoft.CodeAnalysis;
+
+namespace Riok.Mapperly.Symbols;
+
+public record MapperMethodReference(IMethodSymbol Method, INamedTypeSymbol? TargetType, ISymbol? TargetMember);
```

Comment:
Add a comment explaining when which field is set?
Did I get this correctly?
* Neither `TargetType` and `TargetMember` is set if a method of the current mapper is referenced
* `TargetType` is set if a method of an external type is referenced.
* `TargetMember` is set if an instance method of a local member is referenced.